Objects/codeobject.c

```diff
@@ -2089,7 +2097,7 @@ _PyCode_Hydrate(PyCodeObject *code)
 
 /* Hydrate a dehydrated constant.
  *
- * Normally this is done from the LOAD_LAZY_CONSTANT instruction,
+ * Normally this is done from the LAZY_LOAD_CONSTANT instruction,
  * but sometimes we need to do it without an interpreter handy
  * (when fully hydrating co_consts before passing it to Python).
  * In that case, just construct a code object for these and eval it.
```

Comment:
You need redirect support below in eval_constant(). AFAICT this would do it:
```diff
diff --git a/Objects/codeobject.c b/Objects/codeobject.c
index 7173e390c5..c28160da12 100644
--- a/Objects/codeobject.c
+++ b/Objects/codeobject.c
@@ -2118,6 +2118,10 @@ eval_constant(PyCodeObject *code, int index)
         return result;
     }
     uint32_t offset = pyc->const_offsets[index];
+    if (offset & 1) {
+        index = offset >> 1;
+        offset = pyc->const_offsets[index];
+    }
     uint32_t *pointer = (uint32_t *)lazy_get_pointer(pyc, offset);
     uint32_t stacksize = *pointer++;
     Py_ssize_t n_instrs = *pointer++;
```